Foundation/FileManager.swift

```diff
@@ -825,6 +825,8 @@ open class FileManager : NSObject {
         guard let enumerator2 = enumerator(atPath: path2) else {
             return false
         }
+        enumerator1.skipDescendants()
+        enumerator2.skipDescendants()
```

Comment:
In both the Windows and POSIX implementations, `_compareDirectories` and `_contentsEqual` are mutually recursive. Not skipping descendants causes `_contentsEqual` to all `_compareDirectories` on every subdirectory which will each then call `_contentsEqual` on every one of their children causing excessive comparisons. This should fix **a lot** of unneeded recursion.